shop/src/components/payment/crypto/CryptoDirect.js

```diff
@@ -18,6 +18,9 @@ const PayWithCryptoDirect = ({ submit, encryptedData, onChange, loading }) => {
   const [{ cart }, dispatch] = useStateValue()
   const [activeToken, setActiveToken] = useState({})
   const [tokenPrice, setTokenPrice] = useState('')
+  const [walletBalance, setWalletBalance] = useState({
+    isZero: false
+  }) //create an object called walletBalance, assuming for now that the buyer has some ETH in it already. Credit: https://www.geeksforgeeks.org/reactjs-usestate-hook/
```

Comment:
While I appreciate you adding comments, I'd suggest to avoid adding links to generic tutorials. Links to API reference is fine. But this seems out of place.